server/lib/asset-manager/asset-url-generator.js

```diff
@@ -0,0 +1,44 @@
+const logger = require('@financial-times/n-logger').default;
+const nUiManager = require('./n-ui-manager');
+
+const loadAssetHashesJson = path => {
+	try {
+		return require(`${path}`);
+	} catch(err) {
+		/* istanbul ignore next */
+		logger.warn(`${path} not found. Falling back to un-fingerprinted files.`);
+		return {};
+	}
+}
+
+module.exports = (appName, directory, useLocalAppShell) => {
+
+	const assetHashes = loadAssetHashesJson(`${directory}/public/asset-hashes.json`);
+	const nUiAssetHashes = loadAssetHashesJson(`${directory}/public/n-ui-asset-hashes.json`);
+	const nUiReleaseName = nUiManager.getReleaseName(directory);
+	const nUiUnhashedAssetsRoot = useLocalAppShell ? '/${appName}/n-ui/' : `//www.ft.com/__assets/n-ui/cached/${nUiReleaseName}/`;
```

Comment:
should probably add some tests for dev mode. Not in this PR though as it's already epic